### PR TITLE
Rewrote your mod

### PR DESCRIPTION
**Pull request by simplyWiri**
I rewrote your mod for performance. It went from roughly ~80ms lighting comp to ~.2ms. 

I've left notes inside the code for further optimsations, which, are to be honest, pretty much mandatory, look (MapComp_RegenGrid). You will need to compile using a publicised dll. I've left the .sln and csproj to help.

Please, do _**not**_ use this update as an excuse to advertise.

Cheers, 
Wiri

### CODE REVIEW

File: Source/MapComp_Windows.cs [hunk 1]
```diff
@@ -102,13 +113,20 @@ public void RegenGrid()
...
                     }
                 }
             }
+
+            /*
+             * These are awful, you iterate over huge ranges for pretty much nothing.
+             *  This is easily fixable by:
+             *  A. Cache your reflection!
+             *  B. Instead of hacking past these functions, let them run, and use yours as additionals, why make yourself do more work?
+             */
```
Comment (jptrrs) [inline Source/MapComp_Windows.cs:129]: You should have seen the early versions :-D

**Comment by jptrrs** [comment]
This is awesome, Wiri, thanks!! For an amateur like me, this a true lesson!
I need some time to fully review and test this, but I'll defintely include it in a new update. Tell me your steam handle, if you have one, so I can properly add you as a contributor.
First thing that catches the eye: you split my manual Harmony patches into individual classes. I've grown accostumed to build harmony patches that way after I stubled on something that couldn't be made with annotations. In time, it started looking easier and more practical. Is there a performance benefit in splitting them with annotations? Or is it just an organizational thing?

**Comment by simplyWiri** [comment]
It is easier, if a mod is incompatible with yours, for a mod author to look for the function that is erroring, and go directly to the related function & harmony patch which change it.

So, it is just being generous to modders. Generally it is considered good practice. Although you see mods like HAR use manual patching extensively. End of the day it is taste, and these are relatively simple harmony patches, so it can't hurt making it easier for other modders to fix issues.

I'm not pr'ing for credit. Your mod brother.

**Comment by jptrrs** [comment]
That makes sense. I've since learned the thing that got me started manual patching is totally useless anyway, so I might as well follow the crowd.
About the credits, It's your call. But in my way of thinking that's not a privilege, its just fair to give credit to people who worked on anything.

**Comment by jptrrs** [comment]
About that publicised dll: what is that, and why not just use the regular one?

**Comment by simplyWiri** [comment]
https://github.com/CabbageCrow/AssemblyPublicizer

**Comment by jptrrs** [comment]
Oh, ok. Seem pretty straightforward. Thx!

File: Source/Harmony/CompFlickable.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using HarmonyLib;
+using RimWorld;
+using Verse;
+
+namespace OpenTheWindows
+{
+    [HarmonyPatch(typeof(CompFlickable), nameof(CompFlickable.DoFlick))]
+    public static class CompFlickable_DoFlick
+    {
+        public static void PostFix(CompFlickable __instance)
```
Comment (jptrrs) [inline Source/Harmony/CompFlickable.cs:10]: This uppercase F caused this patch to silent fail, and this cascaded to every window not being able to open or close! It's been months and people only complained about minor things, even thoug the issue was huge. It was really hard to track down what was going on, so I just HAD to write this! :-P

Comment (simplyWiri) [reply Source/Harmony/CompFlickable.cs:10]: Story of my life, my bd

**Pull request merged**